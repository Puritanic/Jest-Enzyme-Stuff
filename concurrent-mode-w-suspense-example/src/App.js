import React, { Suspense, useState, useDeferredValue } from "react";
import { wrapPromise } from "./PersonApi";
import { createResource } from "./PersonApi";
import PostResult from "./PostResult";
import ErrorBoundary from "./ErrorBoundary";
import Person from "./Person";
import Num from "./Num";
import { MyButton } from "./MyButton";
import BigPrime from "./BigPrime.jsx";

const initialResource = createResource();

function App() {
  // const [resource, setResource] = useState(() => createResource())
  const [resource, setResource] = useState(initialResource);
  const deferredResource = useDeferredValue(resource, {
    timeoutMs: 5000
  });
  const [n, setN] = useState(10);
  const [postResource, setPostResource] = useState({
    result: {
      read() {
        return null;
      }
    }
  });
  const isStale = deferredResource !== resource;

  return (
    <div className="App">
      <Suspense fallback={<h1>Loading...</h1>}>
        <ErrorBoundary>
          <Person resource={resource}></Person>
          <PostResult resource={postResource}></PostResult>
        </ErrorBoundary>
      </Suspense>

      <Suspense fallback={<h2>Number loading...</h2>}>
        <div style={{ color: isStale ? "crimson" : "black" }}>
          <Num resource={deferredResource}></Num>
        </div>
      </Suspense>
      <MyButton onClick={() => setResource(createResource())}>
        Refresh Data
      </MyButton>
      <BigPrime n={n}></BigPrime>
      <input value={"" + n} onChange={e => setN(parseInt(e.target.value))} />

      <button
        onClick={() => {
          const promise = fetch("https://enrdixyyuxes.x.pipedream.net/", {
            method: "POST",
            body: JSON.stringify({ hello: "World" })
          })
            .then(x => x.json())
            .then(res => {
              console.log(res);
              return res;
            });

          setPostResource({ result: wrapPromise(promise) });
        }}
      >
        Post request!
      </button>
    </div>
  );
}

export default App;
