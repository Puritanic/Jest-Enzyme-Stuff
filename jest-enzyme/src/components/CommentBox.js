import React from 'react';

class CommentBox extends React.Component {
	state = {
		comment: '',
	};

	onChange = e => this.setState({ comment: e.target.value });

	onSubmit = () => {
		const comment = {
			id: this.state.comment.length * Math.random() * 10,
			content: this.state.comment,
			author: 'Anon',
		};
		this.props.saveComment(comment);
		this.setState({ comment: '' });
	};

	render() {
		return (
			<div className="comment__box">
				<textarea
					onChange={this.onChange}
					value={this.state.comment}
					name="comment_field"
					className="comment_field"
					cols="30"
					rows="10"
				/>
				<button className="comment__button" onClick={this.onSubmit}>
					Submit
				</button>
			</div>
		);
	}
}

export default CommentBox;
