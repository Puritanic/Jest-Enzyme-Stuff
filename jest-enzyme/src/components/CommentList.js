import React from 'react';

import Comment from './Comment';

const CommentList = props => {
	return (
		<ul className="comment__list">
			{props.comments &&
				props.comments.map(comment => <Comment key={comment.id} comment={comment} />)}
		</ul>
	);
};

export default CommentList;
