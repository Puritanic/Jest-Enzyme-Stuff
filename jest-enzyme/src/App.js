import React, { Component } from 'react';
import { connect } from 'react-redux';

import './App.css';

import CommentList from './components/CommentList';
import CommentBox from './components/CommentBox';
import { saveComment } from './Redux/actions/commentsActions';

export class App extends Component {
	onSubmit = data => {
		return this.props.saveComment(data);
	};

	render() {
		console.log(this.props);
		return (
			<div className="App">
				<CommentList comments={this.props.comments.comments} />
				<CommentBox saveComment={this.onSubmit} />
			</div>
		);
	}
}

const mapStateToProps = state => ({
	comments: state.comments,
});

const mapDispatchToProps = dispatch => ({
	saveComment: data => dispatch(saveComment(data)),
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(App);
