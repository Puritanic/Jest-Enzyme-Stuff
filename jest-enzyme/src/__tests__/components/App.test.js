/* eslint-disable no-undef */

import React from 'react';
import ReactDOM from 'react-dom';
import CommentList from '../../components/CommentList';
import CommentBox from '../../components/CommentBox';
import { App } from '../../App';

describe('<App />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<App />);
	});

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<App />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('should Match a Snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should render CommentList component', () => {
		expect(wrapper.find(CommentList).length).toBe(1);
	});

	it('should render CommentBox component', () => {
		expect(wrapper.find(CommentBox).length).toBe(1);
	});
});
