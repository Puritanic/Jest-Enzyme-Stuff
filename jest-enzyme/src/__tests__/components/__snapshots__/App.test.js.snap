// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should Match a Snapshot 1`] = `
<div
  className="App"
>
  <CommentList />
  <CommentBox
    saveComment={[Function]}
  />
</div>
`;
